core/lib/contracts/src/lib.rs

```diff
@@ -334,10 +339,16 @@ impl BaseSystemContracts {
             hash,
         };
 
-        let evm_simulator_bytecode =
+        let mut evm_simulator_bytecode =
             read_sys_contract_bytecode("", "EvmInterpreter", ContractLanguage::Yul);
         let evm_simulator_hash = hash_bytecode(&evm_simulator_bytecode);
 
+        let use_evm_simulator =
+            UseEvmSimulator::from_env().expect("USE EVM SIMULATOR FLAG SHOULD BE SET");
+        if !use_evm_simulator.use_evm_simulator {
+            evm_simulator_bytecode = vec![];
+        }
```

Comment:
Maybe:

```rust
let evm_simulator_bytecode = if UseEvmSimulator::from_env().expect("USE EVM SIMULATOR FLAG SHOULD BE SET").use_evm_simulator {
    read_sys_contract_bytecode("", "EvmInterpreter", ContractLanguage::Yul)
} else {
    vec![]
}
```

Reply:
That is not possible, because then we need to calculate the hash of that bytecode, if we try to calculate the hash of an empty bytecode it gives an error when running `zk init`:
`Invalid bytecode: ()`
We need to still have the hash of the actual interpreter bytecode, and then setting that bytecode to an empty vector.